conda/core/prefix_data.py

```diff
@@ -84,6 +100,35 @@ def __init__(
             else context.pip_interop_enabled
         )
 
+    @classmethod
+    def from_name(cls, name: str, **kwargs):
+        if "/" in name or "\\" in name:
+            raise CondaValueError("Environment names cannot contain path separators")
+        try:
+            return cls(locate_prefix_by_name(name))
+        except EnvironmentNameNotFound:
+            cls(name).validate_name()
+            return cls(Path(_first_writable_envs_dir(), name), **kwargs)
+
+    @classmethod
+    def from_context(cls, validate: bool = False):
```

Comment:
This should also get a nice doc string. When I first saw this method in the code itself, I was left wondering, "okay, but which context are you talking about?" Let's make sure the doc string clarifies that so developers can easily find it.